### PR TITLE
Test cleanup for nose

### PR DESCRIPTION
**Pull request by millerdev**
This is the first in a set of two change sets that I'm working on to get nose to be the default test runner. Most of this is adding cleanup so tests don't affect other tests (nose turns this stuff up because it runs tests in a different order than the django test runner does).

I haven't run this locally with the django test runner, so if travis fails I'll fix it and try again...

@snopoke @biyeun

### CODE REVIEW

**Comment by dimagimon** [comment]
There are 6 errors:

* corehq/apps/data_interfaces/tests/tests.py, line 29 - E302 expected 2 blank lines, found 1
* corehq/apps/dropbox/tests/test_dropbox_upload_helper.py, line 2 - F401 'Domain' imported but unused
* corehq/apps/reports/tests/test_pillows_cases.py, line 467 - E303 too many blank lines (2)
* corehq/tests/nose.py, line 121 - F841 local variable 'db' is assigned to but never used
* corehq/util/test_utils.py, line 68 - E231 missing whitespace after ':'
* corehq/util/tests/test_quickcache.py, line 16 - E261 at least two spaces before inline comment

**Comment by millerdev** [comment]
:white_check_mark:

File: corehq/apps/users/tests/create.py [hunk 1]
```diff
@@ -53,15 +58,21 @@ def testCreateCompleteWebUser(self):
...
+            self.assertEqual(couch_user.email, email)
+            couch_user.add_domain_membership('domain1')
+            self.assertEqual(couch_user.domain_memberships[0].domain, 'domain1')
+            couch_user.add_domain_membership('domain2')
+            self.assertEqual(couch_user.domain_memberships[1].domain, 'domain2')
+            django_user = couch_user.get_django_user()
+            self.assertEqual(couch_user.user_id, CouchUser.from_django_user(django_user).user_id)
+        finally:
+            domain1.delete()
+            domain2.delete()
```
Comment (snopoke) [inline corehq/apps/users/tests/create.py]: Sound rather use 'self.addCleanup(domain1.delete)'

File: corehq/messaging/smsbackends/telerivet/tests/test_log_call.py [hunk 2]
```diff
@@ -1,10 +1,10 @@
-from corehq.apps.ivr.tests.util import LogCallTestCase
+import corehq.apps.ivr.tests.util as util
```
Comment (snopoke) [inline corehq/messaging/smsbackends/telerivet/tests/test_log_call.py]: Why is this better?

File: corehq/apps/data_interfaces/tests/tests.py [hunk 3]
```diff
@@ -15,22 +17,30 @@
...
 MISSING_XLSX = 'missing_forms_bulk.xlsx'
 MALFORM_XLSX = 'malformatted_forms_bulk.xlsx'
 WRONG_FILETYPE = 'wrong_file.xyz'
 
 
+def setup():
+    create_domain(DOMAIN_NAME)
+
+
+def teardown():
```
Comment (snopoke) [inline corehq/apps/data_interfaces/tests/tests.py]: These don't seem to be used

File: corehq/form_processor/tests/test_sharding.py [hunk 4]
```diff
@@ -19,7 +20,8 @@ class ShardingTests(TestCase):
 
     @classmethod
     def setUpClass(cls):
-        cls.partion_config = PartitionConfig()
+        with trap_extra_setup(PartitionedDatabaseRequiredError):
```
Comment (snopoke) [inline corehq/form_processor/tests/test_sharding.py]: Why is this necessary? Does nose not respect the `@skipUnless` decorator on the class?

Comment (snopoke) [inline corehq/form_processor/tests/test_sharding.py]: Also, running these tests on a non-sharded DB is meaning less

File: corehq/util/test_utils.py [hunk 5]
```diff
@@ -60,6 +60,27 @@ def trap_extra_setup(*exceptions, **kw):
         raise
 
 
+def softer_assert(func=None):
```
Comment (snopoke) [inline corehq/util/test_utils.py]: Could use [ContextDecorator](https://github.com/dimagi/commcare-hq/blob/0020c7a9c811ecc3f3dfcb2ca182493318774810/corehq/util/decorators.py#L10-L10) to make this simpler

File: corehq/apps/users/tests/create.py (hunk 1 above)
Comment (millerdev) [inline corehq/apps/users/tests/create.py]: great idea, I forgot about `self.addCleanup`

File: corehq/messaging/smsbackends/telerivet/tests/test_log_call.py (hunk 2 above)
Comment (millerdev) [inline corehq/messaging/smsbackends/telerivet/tests/test_log_call.py]: I don't understand it really, but for some reason the old import was causing `LogCallTestCase` to appear as if it was defined inside this module, which was causing [this conditional](https://github.com/dimagi/commcare-hq/blob/a7963baf03e7de2f2124de7c7d229487a9338202/corehq/apps/ivr/tests/util.py#L55) to fail.

see also: commit message on https://github.com/dimagi/commcare-hq/commit/a7963baf03e7de2f2124de7c7d229487a9338202

File: corehq/apps/data_interfaces/tests/tests.py (hunk 3 above)
Comment (millerdev) [inline corehq/apps/data_interfaces/tests/tests.py]: It's a feature of `nose` [*test modules*](http://nose.readthedocs.org/en/latest/writing_tests.html#test-modules)

Comment (snopoke) [inline corehq/apps/data_interfaces/tests/tests.py]: Ah

On 29 February 2016 at 15:37, Daniel Miller <notifications@github.com>
wrote:

> In corehq/apps/data_interfaces/tests/tests.py
> <https://github.com/dimagi/commcare-hq/pull/10522#discussion_r54409576>:
>
> >  MISSING_XLSX = 'missing_forms_bulk.xlsx'
> >  MALFORM_XLSX = 'malformatted_forms_bulk.xlsx'
> >  WRONG_FILETYPE = 'wrong_file.xyz'
> >
> >
> > +def setup():
> > +    create_domain(DOMAIN_NAME)
> > +
> > +
> > +def teardown():
>
> It's a feature of nose *test modules*
> <http://nose.readthedocs.org/en/latest/writing_tests.html#test-modules>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dimagi/commcare-hq/pull/10522/files#r54409576>.
>



-- 
Simon Kelly
Senior Engineer | Dimagi South Africa

File: corehq/messaging/smsbackends/telerivet/tests/test_log_call.py (hunk 2 above)
Comment (snopoke) [inline corehq/messaging/smsbackends/telerivet/tests/test_log_call.py]: That is odd.

On 29 February 2016 at 15:35, Daniel Miller <notifications@github.com>
wrote:

> In corehq/messaging/smsbackends/telerivet/tests/test_log_call.py
> <https://github.com/dimagi/commcare-hq/pull/10522#discussion_r54409343>:
>
> > @@ -1,10 +1,10 @@
> > -from corehq.apps.ivr.tests.util import LogCallTestCase
> > +import corehq.apps.ivr.tests.util as util
>
> I don't understand it really, but for some reason the old import was
> causing LogCallTestCase to appear as if it was defined inside this
> module, which was causing this conditional
> <https://github.com/dimagi/commcare-hq/blob/a7963baf03e7de2f2124de7c7d229487a9338202/corehq/apps/ivr/tests/util.py#L55>
> to fail.
>
> see also: commit message on a7963ba
> <https://github.com/dimagi/commcare-hq/commit/a7963baf03e7de2f2124de7c7d229487a9338202>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dimagi/commcare-hq/pull/10522/files#r54409343>.
>



-- 
Simon Kelly
Senior Engineer | Dimagi South Africa

File: corehq/apps/data_interfaces/tests/tests.py (hunk 3 above)
Comment (snopoke) [inline corehq/apps/data_interfaces/tests/tests.py]: Perhaps a comment would be good to inform others

File: corehq/form_processor/tests/test_sharding.py (hunk 4 above)
Comment (millerdev) [inline corehq/form_processor/tests/test_sharding.py]: heh, I didn't even notice the `skipUnless` decorator. I guess it's not working with nose. I'll check into it.

File: corehq/util/test_utils.py (hunk 5 above)
Comment (millerdev) [inline corehq/util/test_utils.py]: :+1:

**Comment by biyeun** [comment]
@millerdev fyi merge conflicts :(

File: corehq/util/test_utils.py (hunk 5 above)
Comment (millerdev) [inline corehq/util/test_utils.py]: On second thought, that changes the semantics of the decorator. Instead of 
```py
@softer_assert
def test_foo():
    ...
```
it has to be
```py
@softer_assert()
def test_foo():
    ...
```
Would have to use crazy `__new__` tricks in `ContextDecorator` to get around that. If you don't mind I'm going to keep it the way it is.

Comment (snopoke) [inline corehq/util/test_utils.py]: OK sure thing but at least it would be consistent:
```python
@softer_assert()
def test_foo():
    ...

with softer_assert()
    ...
```

**Comment by snopoke** [comment]
@millerdev is this still GTG?

**Comment by millerdev** [comment]
@snopoke yep
